streams/src/main/java/org/apache/kafka/streams/state/internals/Position.java

```diff
@@ -44,10 +44,12 @@ private Position(final Map<String, Map<Integer, Long>> other) {
     }
 
     public Position update(final String topic, final int partition, final long offset) {
-        position.computeIfAbsent(topic, k -> new ConcurrentHashMap<>());
-        final ConcurrentMap<Integer, AtomicLong> topicMap = position.get(topic);
-        topicMap.computeIfAbsent(partition, k -> new AtomicLong(0));
-        topicMap.get(partition).getAndAccumulate(offset, Math::max);
+        if (topic != null) {
+            position.computeIfAbsent(topic, k -> new ConcurrentHashMap<>());
+            final ConcurrentMap<Integer, AtomicLong> topicMap = position.get(topic);
+            topicMap.computeIfAbsent(partition, k -> new AtomicLong(0));
+            topicMap.get(partition).getAndAccumulate(offset, Math::max);
```

Comment:
Computing the max is something I didn't add in the framework PR. I left it up to the caller, but it looks like pretty much every caller would have to ensure it all over the place, so I'll go ahead and update https://github.com/apache/kafka/pull/11557 to preserve these semantics.